src/coreclr/gc/gc.cpp

```diff
@@ -22915,13 +22959,51 @@ void gc_heap::mark_phase (int condemned_gen_number, BOOL mark_only_p)
     reset_card_marking_enumerators();
 #endif // FEATURE_CARD_MARKING_STEALING
 
+#ifdef STRESS_REGIONS
+    heap_segment* gen0_region = generation_start_segment (generation_of (0));
+    while (gen0_region)
+    {
+        size_t gen0_region_size = heap_segment_allocated (gen0_region) - heap_segment_mem (gen0_region);
+        
+        if (gen0_region_size > 0)
+        {
+            uint8_t* region_mid = heap_segment_mem (gen0_region) + (gen0_region_size / 2);
+
+            if ((num_gen0_regions % pinning_seg_interval) == 0)
+            {
+                int align_const = get_alignment_constant (TRUE);
```

Comment:
It would be helpful to put a comment, e.g. "Pin one object at the beginning".

Reply:
added.